src/planwise/client/projects/handlers.cljs

```diff
@@ -111,16 +112,39 @@
     :facilities :facilities
     :transport :facilities-with-demand))
 
+(defn- update-tabs-state [db next-tab]
+  (let [tabs (get-in db [:current :wizard :tabs])
+        new-tabs (into {}
+                       (map (fn [[tab state]]
+                              (if (and (= state :visiting) (not= tab next-tab))
+                                [tab :visited]
+                                [tab state])) tabs))]
+    (assoc-in db [:current :wizard :tabs] new-tabs)))
+
+(defn- update-next-tab-state [db next-tab]
+  (if (= :unvisited (get-in db [:current :wizard :tabs next-tab]))
+    (assoc-in db [:current :wizard :tabs next-tab] :visiting)
+    db))
+
+(defn- update-wizard-state [db next-tab]
+  (-> db
+      (update-next-tab-state next-tab)
+      (update-tabs-state next-tab)))
+
```

Comment:
Changing the model would yield these methods unnecessary: instead of recalculating the state of each tab when the wizard state changes, simply set the new step, and the `reaction`s in the subscriptions should re-calculate the _visited_ state for each tab as needed.